hipe4ml/TrainingAndTesting/plot_utils.py

```diff
@@ -0,0 +1,310 @@
+        axs.tick_params(axis='both', which='both', direction="in")
+        for tick in axs.xaxis.get_minor_ticks():
+            tick.tick1line.set_markersize(0)
+            tick.tick2line.set_markersize(0)
+            tick.label1.set_horizontalalignment('center')
+
+    ax1.cax.colorbar(heatmap1)
+    ax2.cax.colorbar(heatmap2)
+    ax1.cax.toggle_label(True)
+    return fig
+
+
+def plot_bdt_eff(threshold, eff_sig):
+    """Plot the BDT efficiency calculated with the function bdt_efficiency_array() in training_utils.py
+
+    Input
+    -----------------------------------
+    threshold: score threscold array
+    eff_sig: bdt efficiency array
+
+    Output
+    -----------------------------------
+    res: matplotlib object of the bdt efficiency as a function of the threshold score
+
+    """
+    res = plt.figure()
+    plt.plot(threshold, eff_sig, 'r.', label='Signal efficiency')
+    plt.legend()
+    plt.xlabel('BDT Score')
+    plt.ylabel('Efficiency')
+    plt.title('Efficiency vs Score')
+    plt.grid()
+    return res
+
+
+def plot_roc(y_truth, model_decision):
+    """Calculate and plot the roc curve
+
+    Input
+    -------------------------------------
+    y_truth: test set label
+    model decision: predicted score
+
+    Output
+    -------------------------------------
+    res: matplotlib object with the roc curve
+
+    """
+    # Compute ROC curve and area under the curve
+    fpr, tpr, _ = roc_curve(y_truth, model_decision)
+    roc_auc = auc(fpr, tpr)
+    res = plt.figure()
+    plt.plot(fpr, tpr, lw=1, label='ROC (area = %0.4f)' % (roc_auc))
+    plt.plot([0, 1], [0, 1], '--', color=(0.6, 0.6, 0.6), label='Luck')
+    plt.xlim([-0.05, 1.05])
+    plt.ylim([-0.05, 1.05])
+    plt.xlabel('False Positive Rate', fontsize=12)
+    plt.ylabel('True Positive Rate', fontsize=12)
+    plt.legend(loc="lower right")
+    plt.grid()
+    plt.close()
+    return res
+
+
+def plot_feature_imp(df_in, y_lab, model):
+    """Calculate the feature importance using the shap violin plot for each feature. The calculation
+    is performed on a subsample of the input training/test set. The model could be sklearn or xgboost
+
+    Input
+    -------------------------------------------
+    df_in: test set dataframe
+    y_lab: test set label
+    model: trained model
+
+    Output
+    -------------------------------------------
+    res: matplotlib object with shap feature importance
+
+    """
+    subs_bkg = df_in[y_lab == 0].sample(10000)
+    subs_sig = df_in[y_lab == 1].sample(10000)
+    df_subs = pd.concat([subs_bkg, subs_sig]).sample(frac=1.)
+    explainer = shap.TreeExplainer(model)
+    shap_values = explainer.shap_values(df_subs)
+    res = plt.figure()
+    shap.summary_plot(shap_values, df_subs, show=False)
+    return res
+
+
+def plot_precision_recall(y_test, y_score):
+    """ Plot precision recall curve
+
+    Input
+    -------------------------------------
+    y_test: test set label
+    y_score: predicted score
```

Comment:
What type they should be? Also it looks to me that here y_score should be a probability.